app/javascript/pages/components/Browser.jsx

```diff
@@ -82,8 +84,8 @@ class Browser extends React.Component {
         <div className="category-lists">
           <div className="container tight">
             <DataMenu items={items} datasets={this.props.datasets} onMenuClick={this.handleMenuSelectedItem('menuOneSelectedItem')} />
-            {menuOneSelectedItem ? <DataMenu datasets={this.props.datasets} items={items[menuOneSelectedItem].items} onMenuClick={this.handleMenuSelectedItem('menuTwoSelectedItem')} /> : null}
-            {menuTwoSelectedItem ? <DataMenu items={items[menuOneSelectedItem].items[menuTwoSelectedItem].items} onDatasetClick={this.handleDatasetClick()} /> : null}
+            {items.length > 0 && menuOneSelectedItem ? <DataMenu datasets={this.props.datasets} items={items.filter(item => item.menuTitle === menuOneSelectedItem)[0].items} onMenuClick={this.handleMenuSelectedItem('menuTwoSelectedItem')} /> : null}
```

Comment:
87 and 88 are both really long and are covering a lot of ground---I'd recommend adding line breaks along the if/then breaks and the component props to make it easier to parse at a glance.